src/libraries/Microsoft.Extensions.Http/tests/Microsoft.Extensions.Http.Tests/Logging/LoggingUriOutputTests.cs

```diff
@@ -16,156 +16,151 @@ namespace Microsoft.Extensions.Http.Tests.Logging
 {
     public class LoggingUriOutputTests
     {
-        [Fact]
-        public async Task LoggingHttpMessageHandler_LogsAbsoluteUri()
+        public static TheoryData<bool, bool, bool, bool> Handlers_LogAbsoluteUri_Data()
         {
-            // Arrange
-            var sink = new TestSink();
-
-            var serviceCollection = new ServiceCollection();
-            serviceCollection.AddLogging();
-            serviceCollection.AddSingleton<ILoggerFactory>(new TestLoggerFactory(sink, enabled: true));
-
-            serviceCollection
-            .AddHttpClient("test")
-            .ConfigurePrimaryHttpMessageHandler(() => new TestMessageHandler());
-
-            var services = serviceCollection.BuildServiceProvider();
-
-            var client = services.GetRequiredService<IHttpClientFactory>().CreateClient("test");
-
-
-            // Act
-            var request = new HttpRequestMessage(HttpMethod.Get, "http://api.example.com/search?term=Western%20Australia");
-
-            await client.SendAsync(request);
-
-            // Assert
-            var messages = sink.Writes.ToArray();
-
-            var message = Assert.Single(messages.Where(m =>
+            TheoryData<bool, bool, bool, bool> result = new();
+            bool[] booleans = { true, false };
+            bool[] syncApiVals =
+#if NET
+                booleans;
+#else
+                { false };
+#endif
+            foreach (bool syncApi in syncApiVals)
             {
-                return
-                    m.EventId == LoggingHttpMessageHandler.Log.EventIds.RequestStart &&
-                    m.LoggerName == "System.Net.Http.HttpClient.test.ClientHandler";
-            }));
-
-            Assert.Equal("Sending HTTP request GET http://api.example.com/search?term=Western%20Australia", message.Message);
+                foreach (bool logQueryString in booleans)
+                {
+                    foreach (bool absoluteUri in booleans)
+                    {
+                        foreach (bool scopeHandler in booleans)
+                        {
+                            result.Add(syncApi, logQueryString, absoluteUri, scopeHandler);
+                        }
+                    }
+                }
+            }
+            return result;
         }
 
-        [Fact]
-        public async Task LoggingScopeHttpMessageHandler_LogsAbsoluteUri()
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [MemberData(nameof(Handlers_LogAbsoluteUri_Data))]
```

Comment:
```suggestion
        [MemberData(nameof(Handlers_LogExpectedUri_Data))]
```
to align with test name